core/src/syn/parser/expression.rs

```diff
@@ -1,26 +1,49 @@
 //! This module defines the pratt parser for operators.
 
+use std::ops::Bound;
+
 use reblessive::Stk;
 
-use super::mac::unexpected;
+use super::mac::{expected_whitespace, unexpected};
+use crate::sql::Range;
 use crate::sql::{value::TryNeg, Cast, Expression, Number, Operator, Value};
 use crate::syn::error::bail;
-use crate::syn::token::Token;
+use crate::syn::token::{self, Token};
 use crate::syn::{
 	parser::{mac::expected, ParseResult, Parser},
 	token::{t, TokenKind},
 };
 
+/// An enum which defines how strong a operator binds it's operands.
+///
+/// If a binding power is higher the operator is more likely to directly operate on it's
+/// neighbours.
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Ord, PartialOrd)]
+pub enum BindingPower {
+	Base,
+	Or,
+	And,
+	Equality,
+	Relation,
+	AddSub,
+	MulDiv,
+	Power,
```

Comment:
imo it would be really cool to support `1..10**5`, so maybe range should be before power?

Reply:
There is a trade off. `1..10**5` does indeed look reasonable but in for example `1..10 ** (5 * 2)` and   `1..<number> "10"` (which would be allowed as a result of this reordering) this precedence ordering looks less reasonable. I think that the ranges should only bind expressions that are generally written without white-space in between, like paths and prefix operators.